stockflux-securities-master/src/services/SecuritiesService.js

```diff
@@ -1,37 +1,37 @@
-import ValidationError from "./ValidationError";
-
 async function getWindowOptions() {
   const currentWindow = await window.fin.Window.getCurrent();
   return currentWindow.getOptions();
 }
 
-export async function getSecuritiesData() {
+export async function getSecurities() {
   const options = await getWindowOptions();
   const response = await fetch(
     `${options.customData.apiBaseUrl}/admin/securities`
   );
-  const securities = await response.json();
-  return securities;
+  if (response.status === 200) return await response.json();
+  else {
+    throw new Error('Something went wrong while fetching securities.');
+  }
 }
 
 export async function getSecurity(securityId) {
   const options = await getWindowOptions();
   const response = await fetch(
     `${options.customData.apiBaseUrl}/admin/securities/${securityId}`
   );
-  const json = await response.json();
-
-  if (response.ok) {
-    return json;
+  if (response.status === 200) return await response.json();
+  else {
+    throw new Error(
+      `Something went wrong while fetching security ${securityId}.`
+    );
   }
-  throw new ValidationError(json.messages);
 }
 
-export async function postSecurity(security) {
+export async function postSecurity(security, errorCallback, successCallback) {
```

Comment:
I don't think this approach is right. An `async` function effectively returns a promise. When you return something from inside an async function, that's the equivalent of resolving the promise with that value. When you throw an error inside of an async function, that's the same as rejecting the promise with the error you've thrown. Passing success/error callbacks into functions is an older way of doing things asynchronously and is one of the main things that promises were introduced to improve. They should really be mutually exclusive, in that you would use one approach or the other - generally promises (possibly using async/await) is preferred.

In this case, the function shouldn't accept any callback functions as parameters and you should just resolve (in this case, `return`, since it's an async function) what you'd want to pass to the success handler. You should reject (in this case `throw`, since it's an async function) what you'd want to pass to an error handler.

Reply:
Ok, I've added the ValidationError back in.